### PR TITLE
[CAS-515] Basic download functionality

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://stream-io.atlassian.net/browse/CAS-515

### Description

This PR adds a possibility to download a file from the message list.

What's covered:
- User is able to download any file from the message list via standard Android `DownloadManager`

What's not covered (almost everything):
- Persistent storage for downloads with download status (started/completed/failed) and download progress
- Possibility to observe download state in `ChannelController`. 
- UI that reflects the current state of particular download (able to download / downloading is in progress / downloading completed)
- Possibility to cancel ongoing downloads
- Possibility to navigate to downloaded file directly from the message list

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [X] Comparison screenshots added for visual changes
- [X] Reviewers added

https://user-images.githubusercontent.com/9600921/103778724-b682fa00-5043-11eb-820d-4510b590f6d2.mov

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListView.kt [hunk 1]
```diff
@@ -777,6 +791,14 @@ public class MessageListView : ConstraintLayout {
             attachmentClickListener ?: DEFAULT_ATTACHMENT_CLICK_LISTENER
     }
 
+    /**
+     *
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListView.kt:795]: Empty java doc. Is it necessary?

**Review by leandroBorgesFerreira — APPROVED** [review]
🚢 🚢

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListView.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListView.kt:796]: Thanks!

**Review by leandroBorgesFerreira — APPROVED** [review]
🐑

**Pull request merged**